firebase-installations/src/main/java/com/google/firebase/installations/Utils.java

```diff
@@ -78,4 +74,22 @@ private static String encodeFidBase64UrlSafe(byte[] rawValue) {
             Charset.defaultCharset())
         .substring(0, FID_LENGTH);
   }
+
+  private static byte[] getBytesFromUUID(UUID uuid) {
+    ByteBuffer bb = ByteBuffer.wrap(new byte[17]);
+
+    // The first 4 bits with the constant FID header of 0x7 (0b0111) followed by the last 4 random
+    // bits of the UUID.
+    byte fidPrefixWithLast4bitsOfUUID =
```

Comment:
Not sure, but I think, my suggestion would make this code simpler (this code assumes that the constants or of type "byte".

```
public static String createRandomFid() {
  byte[] uuidBytes = getBytesFromUUID(UUID.randomUUID(), new byte[17]);
  uuidBytes[16] = uuidBytes[0];
  uuidBytes[0] = REMOVE_PREFIX_MASK & uuidBytes[0] | FID_4BIT_PREFIX
  return encodeFidBase64UrlSafe(uuidBytes);
}

private static byte[] getBytesFromUUID(UUID uuid, byte[] output) {
  ByteBuffer bb = ByteBuffer.wrap(output);
  bb.putLong(uuid.getMostSignificantBits());
  bb.putLong(uuid.getLeastSignificantBits());
  return bb.array();
}
```

Reply:
Like Masksym mentioned, we had misunderstood when you said copy the bytes. This seems good too. Changed it. Thank you.